### PR TITLE
chore: Migration script allow multiple privileges

### PR DESCRIPTION
**Pull request by sfc-gh-jcieslak**
## Changes
- Add privilege grouping into the migration script

## Next prs
- More test cases and test types
- More mappings

### CODE REVIEW

File: pkg/scripts/migration_script/mappings_grants.go [hunk 1]
```diff
@@ -39,23 +40,37 @@ func HandleGrants(csvInput [][]string) error {
...
+			grant.GrantOn.String(),
+			grant.Name.FullyQualifiedName(),
+			grant.GrantedTo.String(),
+			grant.GranteeName.FullyQualifiedName(),
+		}, "_")
+	})
+}
+
+func MapGrantToModel(grantGroup []sdk.Grant) (accconfig.ResourceModel, error) {
+	// Assuming all grants in the group are the same type and only differ by privileges
```
[the remark this answers is not in the document]
Comment (sfc-gh-jcieslak) [reply pkg/scripts/migration_script/mappings_grants.go:57]: Impossible to have len == 0 in this case

File: pkg/scripts/migration_script/mappings_grants.go [hunk 2]
```diff
@@ -39,23 +40,37 @@ func HandleGrants(csvInput [][]string) error {
...
+	return GroupByProperty(grants, func(grant sdk.Grant) string {
+		return strings.Join([]string{
+			grant.GrantOn.String(),
+			grant.Name.FullyQualifiedName(),
+			grant.GrantedTo.String(),
+			grant.GranteeName.FullyQualifiedName(),
+		}, "_")
+	})
+}
+
```
[the remark this answers is not in the document]
Comment (sfc-gh-jcieslak) [reply pkg/scripts/migration_script/mappings_grants.go:55]: Don't know what this is about (I think it was erroring because of other pr that this one is based on and I fixed it when applied rebase)

File: pkg/scripts/migration_script/mappings_grants.go [hunk 3]
```diff
@@ -39,23 +40,37 @@ func HandleGrants(csvInput [][]string) error {
...
-// TODO(next pr): it should receive []sdk.Grant because there may be a few rows for the same type but different privileges
-// TODO(next pr): there should be a grouping step before this function.
-func MapGrantToModel(grant sdk.Grant) (accconfig.ResourceModel, error) {
+func GroupGrants(grants []sdk.Grant) map[string][]sdk.Grant {
+	return GroupByProperty(grants, func(grant sdk.Grant) string {
+		return strings.Join([]string{
+			grant.GrantOn.String(),
+			grant.Name.FullyQualifiedName(),
+			grant.GrantedTo.String(),
+			grant.GranteeName.FullyQualifiedName(),
```
Comment (sfc-gh-jmichalak) [inline pkg/scripts/migration_script/mappings_grants.go:50]: I think that GrantOption should also be here.

File: pkg/scripts/migration_script/mapping_utils.go [hunk 4]
```diff
@@ -0,0 +1,12 @@
+package main
+
+func GroupByProperty[T any, K comparable](items []T, getProperty func(T) K) map[K][]T {
```
Comment (sfc-gh-jmichalak) [inline pkg/scripts/migration_script/mapping_utils.go]: Nit: let's have basic unit tests for this function.

File: pkg/scripts/migration_script/mappings_grants.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (sfc-gh-jmichalak) [reply pkg/scripts/migration_script/mappings_grants.go:57]: Maybe in this case, but we can be more defensive for future usage, IMO.

File: pkg/scripts/migration_script/mapping_utils.go (hunk 4 above)
Comment (sfc-gh-asawicki) [reply pkg/scripts/migration_script/mapping_utils.go]: +1
Also, shouldn't we add it to our utils? it seems reusable

File: pkg/scripts/migration_script/mappings_grants.go [hunk 5]
```diff
@@ -39,23 +40,37 @@ func HandleGrants(csvInput [][]string) error {
...
+		}, "_")
+	})
+}
+
+func MapGrantToModel(grantGroup []sdk.Grant) (accconfig.ResourceModel, error) {
+	// Assuming all grants in the group are the same type and only differ by privileges
+	grant := grantGroup[0]
+	privileges := collections.Map(grantGroup, func(grant sdk.Grant) string { return grant.Privilege })
+	privilegeListVariable := tfconfig.ListVariable(
```
Comment (sfc-gh-asawicki) [inline pkg/scripts/migration_script/mappings_grants.go]: let's extract in ext instead of building it here.

File: pkg/scripts/migration_script/mapping_utils.go (hunk 4 above)
Comment (sfc-gh-jcieslak) [reply pkg/scripts/migration_script/mapping_utils.go]: Moved to collections and added basic tests

File: pkg/scripts/migration_script/mappings_grants.go (hunk 5 above)
Comment (sfc-gh-jcieslak) [reply pkg/scripts/migration_script/mappings_grants.go]: Done

File: pkg/scripts/migration_script/mappings_grants.go (hunk 3 above)
Comment (sfc-gh-jcieslak) [reply pkg/scripts/migration_script/mappings_grants.go:50]: Yup, you're right.

File: pkg/internal/collections/collection_helpers_test.go [hunk 6]
```diff
@@ -107,3 +107,52 @@ func Test_MapErr(t *testing.T) {
...
+		assert.Len(t, groups["A"], 2)
+		assert.Contains(t, groups["A"], items[0])
+		assert.Contains(t, groups["A"], items[3])
+
+		assert.Len(t, groups["B"], 2)
+		assert.Contains(t, groups["B"], items[1])
+		assert.Contains(t, groups["B"], items[2])
+	})
+
+	t.Run("multi property grouping", func(t *testing.T) {
```
Comment (sfc-gh-jmichalak) [inline pkg/internal/collections/collection_helpers_test.go:140]: Nit: let's add a test case to know about the empty items list just in case.

Comment (sfc-gh-jcieslak) [reply pkg/internal/collections/collection_helpers_test.go:140]: Will do in the next prs, added to my todos
